cql3/statements/select_statement.cc

```diff
@@ -408,11 +408,11 @@ select_statement::do_execute(query_processor& qp,
                 cql3::selection::result_set_builder(*_selection, now,
                         options.get_cql_serialization_format(), *_group_by_cell_indices), std::move(p),
                 [this, page_size, now, timeout](auto& builder, std::unique_ptr<service::pager::query_pager>& p) {
-                    return do_until([&p] {return p->is_exhausted();},
+                    return utils::result_do_until([&p] {return p->is_exhausted();},
```

Comment:
First user of result_do_until. The do_with() indicates a coroutine here (regardless of resultification). It's also the less hot path of aggregate || nonpaged_filtering.

I suggest to push through a patch that converts this code to a coroutine (in a new helper function to not pollute the caller). It can be done in a separate pull request if you like. Then the resultification will be simpler.

Due to all the continuations it will be faster, too, but in a less-important path.